tests/unit/moduleapi/blockedclient.tcl

```diff
@@ -14,4 +14,17 @@ start_server {tags {"modules"}} {
     	r acquire_gil
     	assert_equal {{Blocked client is not supported inside multi}} [r exec]
     }
+
+    test {Locked GIL acquisition from RM_Call} {
+    	assert_equal {Blocked client is not allowed} [r do_rm_call acquire_gil]
+    }
+
+    test {Blocking command are not block the client on RM_Call} {
+    	assert {[r do_rm_call blpop empty_list 10] eq {}}
+        assert {[r do_rm_call brpop empty_list 10] eq {}}
+        assert {[r do_rm_call brpoplpush empty_list1 empty_list2 10] eq {}}
+        assert {[r do_rm_call blmove empty_list1 empty_list2 LEFT LEFT 10] eq {}}
+        assert {[r do_rm_call bzpopmin empty_zset 10] eq {}}
+        assert {[r do_rm_call bzpopmax empty_zset 10] eq {}}
+    }
```

Comment:
since we can't tell the difference between a timeout response and failure to block, maybe we should use invinite timeout for all of these.

also, maybe it's a good idea to put something in the list, and call one of these again, so we see that when there's no need to block, they do work.

Reply:
@oranagra agree with the second point of the comment. Regarding the first, I thought of matching the behavior with multi (because we can not change the multi without a backward compatibility issue). If we do what you suggest it means that RM_Call and MULTI will act differently and has different treatment inside the commands.